### PR TITLE
Withdraw rpcs

### LINKED ISSUE #70: Make withdraw TX broadcast reliable
Pushing transactions to Flashbots protect don't always get picked up and may need to be resent many times.

We can broadcast the signed tx to several endpoints if they are available, like flashbots, Darkflorist RPC & injected provider if available and not in simulation mode.

### PR DESCRIPTION
**Pull request by 0xjimmy**
- Refactor global details about each chain into a single NETWORK object
- Attempt to use DarkFlorist RPC's to send signed transactions to, and then fallback to using the users browser wallet if the RPC either fails or we don't know any RPC for the corresponding network ID.

Fixes #70

### CODE REVIEW

File: app/ts/components/ConfigureFunding.tsx [hunk 1]
```diff
@@ -134,24 +134,55 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
...
 			if (!recipientAddress.value.address) throw 'No recipient provided'
 
-			const [tx] = await signBundle([{ signer: signers.value.burner, transaction: { chainId: provider.value.chainId, from: signers.value.burner.address, to: addressString(recipientAddress.value.address), value: withdrawAmount.value.amount, gasLimit: 21000, type: 2 }}], provider.value.provider, blockInfo.value, getMaxBaseFeeInFutureBlock(blockInfo.value.baseFee, 10n))
-			const payload = JSON.stringify({ jsonrpc: '2.0', method: 'eth_sendPrivateTransaction', params: [{ tx }] })
-			const flashbotsSig = `${await provider.value.authSigner.getAddress()}:${await provider.value.authSigner.signMessage(id(payload))}`
-			const request = await fetch(appSettings.value.relayEndpoint, { method: 'POST', body: payload, headers: { 'Content-Type': 'application/json', 'X-Flashbots-Signature': flashbotsSig } })
-			const response = await request.json()
-			if (response.error !== undefined && response.error !== null) {
-				throw Error(response.error.message)
+			// Worst case scenario, attempt to send via browser wallet if no NETWORK config for chainId or previos error sending to known RPC
```
Comment (KillariDev) [inline app/ts/components/ConfigureFunding.tsx:155]:
```suggestion
			// Worst case scenario, attempt to send via browser wallet if no NETWORK config for chainId or previous error sending to known RPC
```

File: app/ts/components/ConfigureFunding.tsx [hunk 2]
```diff
@@ -134,24 +134,55 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
...
+			if (useBrowserProvider.value === true) {
+				try {
+					const burnerWithBrowserProvider = signers.value.burner.connect(provider.value.provider)
+					const txInput = await burnerWithBrowserProvider.populateTransaction({ chainId: provider.value.chainId, from: signers.value.burner.address, to: addressString(recipientAddress.value.address), gasLimit: 21000, type: 2, value: withdrawAmount.value.amount, maxFeePerGas: withdrawAmount.value.fee / 21000n })
+					const tx = await burnerWithBrowserProvider.signTransaction(txInput)
+					const txHash = await provider.value.provider.send('eth_sendRawTransaction', [tx])
+					return txHash as string
+				} catch (error) {
+					throw error
+				}
```
Comment (KillariDev) [inline app/ts/components/ConfigureFunding.tsx:165]: this block doesn't do anything as it just catches and throws the error?

File: app/ts/components/ConfigureFunding.tsx [hunk 3]
```diff
@@ -134,24 +134,55 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
...
+			}
+
+			// If user is on network that is in NETWORK, send via ethRpc
+			const chainId = provider.value.chainId.toString(10)
+			if (!(chainId in NETWORKS)) {
+				useBrowserProvider.value = true
+				throw 'Unkown network! If you have Interceptor installed and simulation mode on please switch to signing mode and try again.'
```
Comment (KillariDev) [inline app/ts/components/ConfigureFunding.tsx:172]:
```suggestion
				throw 'Unknown network! If you have Interceptor installed and simulation mode on please switch to signing mode and try again.'
```

File: app/ts/components/ConfigureFunding.tsx [hunk 4]
```diff
@@ -134,24 +134,55 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
...
+			}
+
+			const fundingWithProvider = signers.value.burner.connect(new JsonRpcProvider(NETWORKS[chainId].rpcUrl))
+			try {
+				const tx = await fundingWithProvider.sendTransaction({ chainId: provider.value.chainId, from: signers.value.burner.address, to: addressString(recipientAddress.value.address), gasLimit: 21000, type: 2, value: withdrawAmount.value.amount, maxFeePerGas: withdrawAmount.value.fee / 21000n })
```
Comment (KillariDev) [inline app/ts/components/ConfigureFunding.tsx:177]: hmm what is this calculation?
`maxFeePerGas: withdrawAmount.value.fee / 21000n`

File: app/ts/components/ConfigureFunding.tsx [hunk 5]
```diff
@@ -134,24 +134,55 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
...
+				fundingWithProvider.provider?.destroy()
+				return tx.hash
+			} catch (error) {
+				console.warn('Error sending burner withdraw tx to known RPC:', error)
+				fundingWithProvider.provider?.destroy()
+				useBrowserProvider.value = true
+				throw 'Unkown network! If you have Interceptor installed and simulation mode on please switch to signing mode and try again.'
```
Comment (KillariDev) [inline app/ts/components/ConfigureFunding.tsx:184]:
```suggestion
				throw 'Unknown network! If you have Interceptor installed and simulation mode on please switch to signing mode and try again.'
```

**Review by KillariDev — APPROVED** [review]

File: app/ts/constants.ts [hunk 6]
```diff
@@ -1,7 +1,4 @@
...
-export const MEV_RELAY_GOERLI = 'https://relay-goerli.dark.florist/'
-
-export const BLOCK_EXPLORERS: { [chainId: string]: string } = {
-	'1': 'https://etherscan.io/',
-	'5': 'https://goerli.etherscan.io/'
-}
+export const NETWORKS: { [chainId: string]: { blockExplorer: string, mevRelay: string, rpcUrl: string } } = {
+	'1': { mevRelay: 'https://relay.dark.florist', blockExplorer: 'https://etherscan.io/', rpcUrl: 'https://rpc.dark.florist/flipcardtrustone' },
+	'5': { mevRelay: 'https://relay-goerli.dark.florist', blockExplorer: 'https://goerli.etherscan.io/', rpcUrl: 'https://rpc-goerli.dark.florist/flipcardtrustone' }
+} as const
```
Comment (jubalm) [inline app/ts/constants.ts:4]: you can do without the definition in `NETWORKS` is cast with `const`
```suggestion
export const NETWORKS = {
	'1': { mevRelay: 'https://relay.dark.florist', blockExplorer: 'https://etherscan.io/', rpcUrl: 'https://rpc.dark.florist/flipcardtrustone' },
	'5': { mevRelay: 'https://relay-goerli.dark.florist', blockExplorer: 'https://goerli.etherscan.io/', rpcUrl: 'https://rpc-goerli.dark.florist/flipcardtrustone' }
} as const
```

**Review by jubalm — APPROVED** [review]

File: app/ts/components/ConfigureFunding.tsx [hunk 7]
```diff
@@ -127,31 +128,63 @@ const WithdrawModal = ({ display, blockInfo, signers, appSettings, provider }: {
 	}
 
 	const withdrawAmount = useComputed(() => {
-		let fee = (blockInfo.value.baseFee + blockInfo.value.priorityFee) * 11n / 10n * 21000n
+		let maxFeePerGas = getMaxBaseFeeInFutureBlock(blockInfo.value.baseFee, 20n) + blockInfo.value.priorityFee;
```
Comment (0xjimmy) [inline app/ts/components/ConfigureFunding.tsx:131]:
```suggestion
		let maxFeePerGas = getMaxBaseFeeInFutureBlock(blockInfo.value.baseFee, 5n) + blockInfo.value.priorityFee;
```

Allow for max 60 seconds of worse case gas increases.

**Pull request merged**